lib/vfscore/vfs.h

```diff
@@ -69,89 +84,819 @@ extern int vfs_debug;
+ *	Pointer to task containing the cwd
+ * @param path
+ *	The target path
+ * @param mode
+ *	Unused parameter
+ * @param[out] full
+ *	Full path to be returned
+ * @return
+ *	- (0):  Successfully initialized
+ *	- (<0): Negative value with error code
+ */
+int task_conv(struct task *t, const char *path, int mode, char *full);
+
+/**
+ * Converts to full path from the wd of task and cpath.
+ *
+ * @param wd
+ *	Working directory
+ * @param cpath
+ *	The target path
+ * @param[out] full
+ *	Full path to be returned
+ * @return
+ *	- (0):  Successfully initialized
+ *	- (<0): Negative value with error code
+ */
+int path_conv(char *wd, const char *cpath, char *full);
+
+/**
+ * This function is not implemented or used at this point.
+ */
+int sec_vnode_permission(char *path);
+
+/**
+ * Converts a pathname into a pointer to a dentry.
+ *
+ * @param path
+ *	The full path name
+ * @param[out] dpp
+ *	Dentry to be returned
+ * @return
+ *	- (0):  Successfully initialized
+ *	- (<0): Negative value with error code
+ */
+int namei(const char *path, struct dentry **dpp);
 
+/**
+ * Converts the last component in the path to a pointer to a dentry.
+ *
+ * @param path
+ *	The full path name
+ * @param ddp
+ *	Pointer to dentry of parent
+ * @param[out] dp
+ *	Dentry to be returned
+ * @return
+ *	- (0):  Successfully initialized
+ *	- (<0): Negative value with error code
+ */
+int namei_last_nofollow(char *path, struct dentry *ddp, struct dentry **dp);
+
+/**
+ * Searches a pathname.
+ * This routine returns a locked directory vnode and file name.
+ *
+ * @param path
+ *	Full path to the file
+ * @param[out] dpp
+ *	Pointer to the dentry for the directory
+ * @param[out] name
+ *	If non-null, pointer to file name in path
+ * @return
+ *	- (0):  Successfully initialized
+ *	- (<0): Negative value with error code
+ */
+int lookup(char *path, struct dentry **dpp, char **name);
+
+/**
+ * Initializes the vnode buckets of the vnode table.
+ * It is called once (from vfscore_init()) in initialization.
+ *
+ */
+void vnode_init(void);
+
+/**
+ * Calls dentry_init.
+ *
+ */
+void lookup_init(void);
+
+/**
+ * Gets the root directory and mount point for specified path.
+ *
+ * @param path
+ *	Full path
+ * @param[out] mp
+ *	Pointer to the mount point
+ * @param[out] root
+ *	Pointer to root directory in path
+ * @return
+ *	- (0):  Successfully initialized
+ *	- (<0): Negative value with error code
+ */
+int vfs_findroot(const char *path, struct mount **mp, char **root);
+
+/**
+ * Copies size bytes from the src string into the destination
+ * with bounds-checking.
+ *
+ * @param dest
+ *	Destination string
+ * @param src
+ *	Source string
+ * @param size
+ *	Number of bytes to be copied
+ * @return
+ *	- (0):  Successfully initialized
+ *	- (<0): Negative value with error code
+ */
+int vfs_dname_copy(char *dest, const char *src, size_t size);
+
+/**
+ * This function is not used at this point.
+ *
+ * @return
+ *	Always returns 0
+ */
+int fs_noop(void);
+
```

Comment:
Isn't this supposed to be a just `nop` function? Is so should we just say that is a `nop` function that always returns 0?

Reply:
I tried to solve [the comment](https://github.com/unikraft/unikraft/pull/707#discussion_r1070288977) from @razvand